tests/e2e/karydia_default_network_policy_test.go

```diff
@@ -0,0 +1,151 @@
+				t.Fatalf("Exit status with unexpected code: %d %s", exitCode, command)
+			}
+		} else {
+			t.Fatalf("Could not get exit code.")
+		}
+	} else {
+		// success, exitCode should be 0 if go is ok
+		ws := cmd.ProcessState.Sys().(syscall.WaitStatus)
+		exitCode = ws.ExitStatus()
+
+		if exitCode != expectedExitCode {
+			t.Fatalf("Exit status with unexpected code: %d %s", exitCode, command)
+		}
+	}
+}
+
+func TestNetworkPolicyLevel1(t *testing.T) {
+
+	var namespace *corev1.Namespace
+	var err error
+
+	namespace, err = f.CreateTestNamespace()
+	if err != nil {
+		t.Fatalf("failed to create test namespace: %v", err)
+	}
+
+	ns := namespace.ObjectMeta.Name
+
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "karydia-network-policy-1",
+			Namespace: ns,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  "alpine",
+					Image: "alpine",
+					Args: []string{
+						"/bin/sh",
+						"-c",
+						"sleep 60m",
+					},
+				},
+			},
+		},
+	}
+
+	createdPod, err := f.KubeClientset.CoreV1().Pods(ns).Create(pod)
+	if err != nil {
+		t.Fatalf("Failed to create: " + err.Error())
+	}
+	podName := createdPod.ObjectMeta.Name
+
+	timeout := 2 * time.Minute
+	if err := f.WaitPodRunning(ns, podName, timeout); err != nil {
+		t.Fatalf("pod never reached state running")
+	}
+
+	// Test Host network
+	cmd1 := "kubectl exec -it --namespace=" + ns + " " + podName + " -- wget --spider --timeout 3 10.250.0.0"
+	execCommandAssertExitCode(t, cmd1, TimeOut)
+
+	cmd2 := "kubectl exec -it --namespace=" + ns + " " + podName + " -- wget --spider --timeout 3 10.250.1.1"
+	execCommandAssertExitCode(t, cmd2, TimeOut)
+
+	cmd3 := "kubectl exec -it --namespace=" + ns + " " + podName + " -- wget --spider --timeout 3 10.250.50.200"
+	execCommandAssertExitCode(t, cmd3, TimeOut)
+
+	// Meta Data Servicess
+	cmd4 := "kubectl exec -it --namespace=" + ns + " " + podName + " -- wget --spider --timeout 3 169.254.169.254"
+	execCommandAssertExitCode(t, cmd4, TimeOut)
+
+	cmd5 := "kubectl exec -it --namespace=" + ns + " " + podName + " -- wget --spider --timeout 3 169.254.0.0"
+	execCommandAssertExitCode(t, cmd5, TimeOut)
+
+	cmd6 := "kubectl exec -it --namespace=" + ns + " " + podName + " -- wget --spider --timeout 3 169.254.2.2"
+	execCommandAssertExitCode(t, cmd6, TimeOut)
+
+	// Alibaba Clouds
+	cmd7 := "kubectl exec -it --namespace=" + ns + " " + podName + " -- wget --spider --timeout 3 100.100.0.0"
+	execCommandAssertExitCode(t, cmd7, TimeOut)
+
+	cmd8 := "kubectl exec -it --namespace=" + ns + " " + podName + " -- wget --spider --timeout 3 100.100.1.3"
+	execCommandAssertExitCode(t, cmd8, TimeOut)
+
+	cmd9 := "kubectl exec -it --namespace=" + ns + " " + podName + " -- wget --spider --timeout 3 100.100.60.155"
+	execCommandAssertExitCode(t, cmd9, TimeOut)
+
+	// External traffic
+	cmd10 := "kubectl exec -it --namespace=" + ns + " " + podName + " -- wget --spider --timeout 3 www.google.de"
```

Comment:
We should use static / fixed IPs for the success test cases, as well, instead of URLs which need to be resolved by a DNS first.

Reply:
I added test cases.